rpe/extractors/pubsub_metadata.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2021 The resource-policy-evaluation-library Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional
+
+from google.pubsub_v1 import PubsubMessage
+import jmespath
+
+
+@dataclass
+class PubsubMessageMetadata:
+    publish_time: datetime
+    message_id: str
+    attributes: str
+    message_metadata: Optional[dict]
+
+
+def get_pubsub_message_metadata(message: PubsubMessage, message_metadata):
```

Comment:
Can we remove the second argument here. This seems very specific to this extractor. The extractor can merge the metadata retrieved by this call and whatever else it wants